packages/vite/src/node/plugins/esbuild.ts

```diff
@@ -243,20 +258,55 @@ export const buildEsbuildPlugin = (config: ResolvedConfig): Plugin => {
         return null
       }
 
-      const res = await transformWithEsbuild(code, chunk.fileName, {
+      const isEsLibBuild = config.build.lib && opts.format === 'es'
+      let options: TransformOptions = {
         ...config.esbuild,
         target: target || undefined,
-        ...(minify
-          ? {
-              // Do not minify ES lib output since that would remove pure annotations
-              // and break tree-shaking
-              // https://github.com/vuejs/core/issues/2860#issuecomment-926882793
-              minify: !(config.build.lib && opts.format === 'es'),
-              treeShaking: true,
-              format: rollupToEsbuildFormatMap[opts.format]
-            }
-          : undefined)
-      })
+        format: rollupToEsbuildFormatMap[opts.format]
```

Comment:
```suggestion
        format: rollupToEsbuildFormatMap[opts.format],
        minify: false
```
Maybe we should set this here and only override it when https://github.com/vitejs/vite/pull/8754/files#diff-6d149ac9706cd508b52db0c059a0f01d670620a9a5a73cd7b122ce62b6b69471R292-R296?

Or we should set `{ ...options, minify: false, treeShaking: true }` here.
https://github.com/vitejs/vite/pull/8754/files#diff-6d149ac9706cd508b52db0c059a0f01d670620a9a5a73cd7b122ce62b6b69471R275

Reply:
I'd prefer setting `minify` inside the `if` conditions so it's easier to follow, which would be the second choice you suggested. I think your suggestion make sense after we remove the `esbuild.minify` type. If they specify it true for some reason, it shouldn't change the minify logic.

Reply:
Done in 69d81f3